### PR TITLE
samples: nrf_desktop: Keep keyboard up while being used

### PR DESCRIPTION
**Pull request by pdunaj**
Reset power down counter on key strokes.
Also reset the counter in case of BLE connectivity event.

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

File: samples/nrf_desktop/src/modules/power_manager.c [hunk 1]
```diff
@@ -241,41 +245,42 @@ static bool event_handler(const struct event_header *eh)
...
+				struct wake_up_event *wue =
+					new_wake_up_event();
+				EVENT_SUBMIT(wue);
 			}
-			return false;
+			break;
+
+		case USB_STATE_DISCONNECTED:
+			usb_connected = false;
+			power_down_counter_reset();
```
Comment (Qbicz) [inline samples/nrf_desktop/src/modules/power_manager.c]: Good idea to reset the counter on USB disconnect.

Comment (pdunaj) [reply samples/nrf_desktop/src/modules/power_manager.c]: yep

Comment (wiminordic) [reply samples/nrf_desktop/src/modules/power_manager.c]: nrf_desktop CI is failing with power_manager.c:142 assert. Should not be merged until this is fixed.

Comment (Qbicz) [reply samples/nrf_desktop/src/modules/power_manager.c]: Looks like a `break` is missing after `POWER_STATE_OFF`. Didn't notice that before.

Comment (pdunaj) [reply samples/nrf_desktop/src/modules/power_manager.c]: break is not missing here (at least it was not added intentionally). This (now common) code was not used in this path before. Let me check if states are right.

Comment (pdunaj) [reply samples/nrf_desktop/src/modules/power_manager.c]: Fixed

**Comment by pdunaj** [comment]
Hi @wiminordic , @jhn-nordic , could you give me a green tick on this issue?

**Comment by pdunaj** [comment]
Hi @pizi-nordic , or maybe you can just merge it :)

**Pull request merged**